deploy/vagrant/bin/spot_request.py

```diff
@@ -203,12 +203,16 @@ def main(args):
         info('getting instance IPs...')
         for r in requests:
             instance_id = r.instance_id
+            info('waiting for ip to be allocated to the machine')
             ip = conn.get_only_instances([instance_id])[0].ip_address
```

Comment:
I have a general question: does get_only_instances() always return an array with at least 1 element?

Is it guaranteed for the ip_address at some time to not be None?